Formula/l/lbdb.rb

```diff
@@ -29,5 +30,7 @@ def install
 
   test do
     assert_match version.major_minor.to_s, shell_output("#{bin}/lbdbq -v")
+    assert_predicate lib/"lbdb/m_abook", :exist?, "m_abook module is missing!"
+    assert_predicate lib/"lbdb/m_khard", :exist?, "m_khard module is missing!"
```

Comment:
These tests are better than nothing, but can we try a very simple test of functionality?

If doing it properly takes too much setup, we can try using these modules erroneously so that we can check that we get the expected error code(s)/message(s).

Reply:
The module is just a basic bash script calling a third party tool (khard, installed as a deop) for doing a research in the local khard databases that would have to be populated with addresses to be tested, please have a look at the module test: https://github.com/RolandRosenfeld/lbdb/blob/master/m_khard.sh.in
Honestly, I'm not sure I've the time and interest for writing tests that would be longer than the module itself...

Reply:
I personally find that for the case of this particular PR having a check that the module is indeed present is enough. Especially as before it wasn't present.

Reply:
Anything like `lbdb config` that we could check for the `m_khard` string?

Reply:
Nope I don't think, AFAIU `lbdbq` the query command will try to dynamically look for modules in the module path (so either the default one from the package, or a custom one in the configuration), and will try to open a command specific to this module name (looking for `m_khard_query` for the `m_khard` module, `m_abook_query` for the `m_abook` module and so on). AFAICS modules are always shell script with a `something_query` command, so no need to have the code include much other stuff as it can easily/dynamically load any custom module.
You can look at the changes made to add the khard support to lbdb: https://github.com/RolandRosenfeld/lbdb/commit/8bb93e5b7379a49bd8b0fed30a3bc1db4c3ce476, the only relevant thing is really the module itself, I don't think it makes much sense to look for something else like a comment in the RC file template.

Reply:
AFAICS you don't have to rebuild lbdb to add a new custom module, you just add it to the module path, following the convention/template/example for creating a module.

Reply:
So would checking the output of `lbdbq m_khard` be better then? I'm just trying to make sure the test still functions if the filename changes. Or for it to fail correctly if the permissions are wrong.

Reply:
This will only try to look for `m_khard` in the configured address books...
I understand you want to use my time to improve the tests, but I don't have much ideas about how to improve it more than what is already being done (checking the module file was created) and I also don't have that much time 😅 
To me it looks like more advanced test case like the one you seem to be willing may better be added to the lbdb repo itself. Here we are just expecting a third party tool to build its files properly, I'm not sure if we should try to overcontrol things, especially as we are not doing any custom patching or weird things, it's just relying on the standard configure + building from the upstream.
We could check the permissions on the modules, but I'm also a bit unsure it's really worth the case...

Reply:
```
❯ lbdbq m_khard
lbdbq: no matches
```

That seems promising, what will `lbdbq notarealthing` do? Will it give a different error? Because then we have our test case.

Reply:
> To me it looks like more advanced test case like the one you seem to be willing may better be added to the lbdb repo itself. 

We're really just looking for a command that a user would run to see if it works. That way we can rule out some problems already when people file an issue.
> Here we are just expecting a third party tool to build its files properly, I'm not sure if we should try to overcontrol things, especially as we are not doing any custom patching or weird things, it's just relying on the standard configure + building from the upstream.

We do, but sometimes there are issues that only arise when someone builds in a non-standard location like `/opt/homebrew` and it's better to catch those in CI.
> We could check the permissions on the modules, but I'm also a bit unsure it's really worth the case..

We have no assurance that that's static, this is why I'm suggesting commands. Those are usually stable, while I would consider module filenames and permissions an internal implementation detail.

Reply:
> ```
> ❯ lbdbq m_khard
> lbdbq: no matches
> ```
> 
> That seems promising, what will `lbdbq notarealthing` do? Will it give a different error? Because then we have our test case.

As said, here m_khard is just treated a random search parameter, that lbdb will search for in the address books/sources available by the configured plugins:
```shell
lbdbq plop
lbdbq: no matches
❯ lbdbq noreply
lbdbq: 2 matches
author@noreply.github.com       Author  pref
bdii@noreply.github.com EGI-Foundation/bdii     internet
```

> We're really just looking for a command that a user would run to see if it works. That way we can rule out some problems already when people file an issue.
I think the only way to ensure it works is having a `khard` setup mostly working, with some existing addresses created locally, and checking that `lbdbq` can query it. But just the `khard` setup would be a bit complex, and yes we could create a contact in khard and search for it, it would be quite a lot of work/effort/test/resource consumption/wasting for just this. As you can see the module is just a small shell wrapper calling khard directly with the single provider parameter: https://github.com/RolandRosenfeld/lbdb/blob/master/m_khard.sh.in.

Reply:
Let's just merge this as-is. Thanks!

Reply:
Thanks!